Container/Container.cs

```diff
@@ -37,9 +37,22 @@ public SimpleDependencyContainer()
         /// Resolves a dependency
         /// </summary>
         /// <param name="type"></param>
+        /// <param name="alreadyResolvedList"></param>
         /// <returns></returns>
-        public object Resolve(Type type)
+        private object Resolve(Type type, List<Type> alreadyResolvedList)
```

Comment:
Might be better to use a hashset since we're using the contains function (O(n) in list while O(1) in hashset)